code/__HELPERS/unique_ids.dm

```diff
@@ -0,0 +1,51 @@
+// Unique Datum Identifiers
+
+// Basically, a replacement for plain REF()s that ensure the reference still
+// points to the exact same datum/client, but doesn't prevent GC like tags do.
+
+// An unintended side effect of the way UIDs are formatted is that the locate()
+// proc will ignore the number and attempt to locate the reference. I consider
+// this a feature, since it means they're conveniently backwards compatible.
+
+// Turns this:
+//   var/myref = "[REF(mydatum)]"
+//   var/datum/D = locate(myref)
+// into this:
+//   var/myUID = mydatum.UID()
+//   var/datum/D = locateUID(myUID)
+
+GLOBAL_VAR_INIT(next_unique_datum_id, 1)
+
+/datum/var/tmp/unique_datum_id = null
+//client/var/tmp/unique_datum_id = null
+
+/datum/proc/UID()
+	if(!unique_datum_id)
+		var/tag_backup = tag
+		tag = null // Grab the raw ref, not the tag
+		unique_datum_id = "[REF(src)]_[GLOB.next_unique_datum_id++]"
+		tag = tag_backup
+	return unique_datum_id
+
+/*/client/proc/UID()
+	if(!unique_datum_id)
+		unique_datum_id = "[REF(src)]_[next_unique_datum_id++]"
+	return unique_datum_id*/
```

Comment:
Why're we adding commented out code? Can we remove this, unless there is a great reason for it I'm missing?

Reply:
I mean if/when we do a full UID() port this will be relevant but I guess it can be removed until then.